### PR TITLE
Add warning for Unknown Reference in Variable Characterisations

### PR DESCRIPTION
**Pull request by Entenwilli**
This PR adds a logger warning to indicate an issue, if a reference to a variable in the SEFF could to be found. This PR fixes the issue 
#41

### CODE REVIEW

File: bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/entity/pcm/CallingUserActionSequenceElement.java [hunk 1]
```diff
@@ -56,12 +58,37 @@ public AbstractActionSequenceElement<EntryLevelSystemCall> evaluateDataFlow(List
...
        return new CallingUserActionSequenceElement(this, characteristicsCalculator.getCalculatedCharacteristics(), nodeCharacteristics);
     }
+    
+    private void checkParameter(List<VariableCharacterisation> variableCharacterisations) {
+    	List<String> parameter = 
+    			this.getElement().getOperationSignature__EntryLevelSystemCall().getParameters__OperationSignature().stream()
+    			.map(it -> it.getParameterName())
+    			.toList();
+    	
+    	List<String> referencedParameter =
```
Comment (sebinside) [inline bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/entity/pcm/CallingUserActionSequenceElement.java:79]: The second part of this looks a lot like code duplication. Of course, the first part has to be specific to ExternalCalls / EntryLevelSystemCalls, but the second is the same, right?

File: bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/entity/pcm/CallingUserActionSequenceElement.java [hunk 2]
```diff
@@ -56,12 +58,37 @@ public AbstractActionSequenceElement<EntryLevelSystemCall> evaluateDataFlow(List
...
+    			variableCharacterisations.stream()
+    			.map(it -> it.getVariableUsage_VariableCharacterisation().getNamedReference__VariableUsage().getReferenceName())
+    			.toList();
+    	
+    	referencedParameter.stream()
+    	.filter(it -> !parameter.contains(it))
+    	.forEach(it -> {
+    		logger.warn("Unknown reference to variable " + it + " in variable characterisation in element " + this.toString());
+    		logger.warn("Present variables:" + parameter + ", Referenced parameter: " + referencedParameter);
+    		throw new IllegalStateException();
```
Comment (sebinside) [inline bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/entity/pcm/CallingUserActionSequenceElement.java:89]: Why does this throw an exception but the upper implementation of ExternalCalls does not?

File: bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/pcm/PCMDataCharacteristicsCalculator.java [hunk 3]
```diff
@@ -7,6 +7,7 @@
 import java.util.stream.Collectors;
 import java.util.Comparator;
 
+import org.apache.log4j.Logger;
```
Comment (sebinside) [inline bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/pcm/PCMDataCharacteristicsCalculator.java:10]: Unused import? At least from what I can see here on github

**Review by sebinside — CHANGES_REQUESTED** [review]
Functionality-wise, it works fine, thank you. Here are only some minor comments regarding the code quality.

File: bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/entity/pcm/CallingUserActionSequenceElement.java (hunk 1 above)
Comment (Entenwilli) [reply bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/entity/pcm/CallingUserActionSequenceElement.java]: I've moved the duplicate code into the `PCMDataCharacteristicsCalculator` to remove the code duplication here

File: bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/entity/pcm/CallingUserActionSequenceElement.java (hunk 2 above)
Comment (Entenwilli) [reply bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/entity/pcm/CallingUserActionSequenceElement.java]: This has left to debug the warnings, but I forgot to remove it. I've removed it now

File: bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/pcm/PCMDataCharacteristicsCalculator.java (hunk 3 above)
Comment (Entenwilli) [reply bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/pcm/PCMDataCharacteristicsCalculator.java:10]: As the duplicate code uses the logger now, this import is now used again

**Review by sebinside — APPROVED** [review]

**Review by sebinside — APPROVED** [review]

**Pull request merged**